src/test/java/com/exasol/adapter/dialects/elasticsearch/ElasticSearchSqlDialectIT.java

```diff
@@ -0,0 +1,332 @@
+package com.exasol.adapter.dialects.elasticsearch;
+
+import static com.exasol.adapter.dialects.elasticsearch.ITConfiguration.*;
+import static com.exasol.matcher.ResultSetStructureMatcher.table;
+import static javax.json.Json.createObjectBuilder;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.IOException;
+import java.sql.*;
+import java.util.concurrent.TimeoutException;
+
+import javax.json.JsonObject;
+
+import org.hamcrest.Matcher;
+import org.hamcrest.core.IsNull;
+import org.junit.jupiter.api.*;
+import org.testcontainers.containers.JdbcDatabaseContainer.NoDriverFoundException;
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import com.exasol.bucketfs.Bucket;
+import com.exasol.bucketfs.BucketAccessException;
+import com.exasol.containers.ExasolContainer;
+import com.exasol.dbbuilder.dialects.DatabaseObject;
+import com.exasol.dbbuilder.dialects.exasol.*;
+import com.exasol.dbbuilder.dialects.exasol.AdapterScript.Language;
+
+@Tag("integration")
+@Testcontainers
+class ElasticSearchSqlDialectIT {
+    @Container
+    private static final ExasolContainer<? extends ExasolContainer<?>> EXASOL = new ExasolContainer<>(
+            getExasolDockerImageReference()).withReuse(true);
+    @Container
+    private static final ElasticsearchContainer ES_CONTAINER = new ElasticsearchContainer(
+            ELASTICSEARCH_DOCKER_IMAGE_REFERENCE).withReuse(true);
+    private static final String ES_DIALECT_NAME = "ES";
+    private static final String VIRTUAL_SCHEMA_NAME = "VIRTUAL_SCHEMA_ES";
+    private static final String INDEX_NAME = "index";
+    private static Connection connection;
+    private static ExasolObjectFactory objectFactory;
+    private static ExasolSchema adapterSchema;
+    private static AdapterScript adapterScript;
+    private static VirtualSchema virtualSchema;
+    private static ConnectionDefinition jdbcConnection;
+    private static ElasticSearchGateway esGateway;
+
+    @BeforeAll
+    static void beforeAll() throws BucketAccessException, InterruptedException, TimeoutException, IOException,
+            NoDriverFoundException, SQLException {
+        connection = EXASOL.createConnection("");
+        objectFactory = new ExasolObjectFactory(connection);
+        adapterSchema = objectFactory.createSchema("ADAPTER_SCHEMA");
+        adapterScript = installVirtualSchemaAdapter(adapterSchema);
+        ES_CONTAINER.start();
+        esGateway = ElasticSearchGateway.connectTo(ES_CONTAINER.getHttpHostAddress());
+        esGateway.startTrial();
+        esGateway.closeConnection();
+    }
+
+    private static AdapterScript installVirtualSchemaAdapter(final ExasolSchema adapterSchema)
```

Comment:
as long as you only have one integration test it's ok to have helper methods like this one here. In order to use abstract test from common JDBC, you will however have to move them to a separate class in order to avoid code duplication